packages/core/src/services/scales-cartesian.ts

```diff
@@ -118,6 +142,64 @@ export class CartesianScales extends Service {
 		}
 	}
 
+	isDualAxes() {
+		return this.isDualAxes;
+	}
+
+	// if any of the axes objects have datasets [] asserted we flag the chart as dual axes
+	// it does not count as dual axes if it just has another axis turned on but is not actually using it to map a dataset
+	determineDualAxes() {
+		const options = this.model.getOptions();
+		const axesOptions = Tools.getProperty(options, "axes");
+
+		if (axesOptions[AxisPositions.LEFT]?.datasets  &&  axesOptions[AxisPositions.RIGHT]
+			|| axesOptions[AxisPositions.RIGHT]?.datasets  &&  axesOptions[AxisPositions.LEFT]
+			|| axesOptions[AxisPositions.TOP]?.datasets  &&  axesOptions[AxisPositions.BOTTOM]
+			|| axesOptions[AxisPositions.BOTTOM]?.datasets  &&  axesOptions[AxisPositions.TOP]) {
+				this.dualAxes = true;
+		}
+	}
+
+	/**
+	 * Returns the position of the range axis for the data item or the datagroups provided
+	 * @param datum
+	 * @param groups
+	 */
+	getDualAxesRangePosition(datum, groups?) {
+		const options = this.model.getOptions();
+		const { groupMapsTo } = options.data;
+		const axisOptions = Tools.getProperty(options, "axes", this.secondaryRangeAxisPosition);
+		let dataset;
+		if (datum) {
+			dataset = datum[groupMapsTo];
+		} else if (groups && groups.length > 0) {
+			dataset = groups[0];
+		} else {
+			return this.rangeAxisPosition;
+		}
+
+		if (axisOptions?.datasets && axisOptions.datasets.includes(dataset)) {
+			return this.secondaryRangeAxisPosition;
+		}
+		return this.rangeAxisPosition;
+	}
+
+
+	/**
+	 * Returns the position of the domain axis that the datum is mapped with
+	 * @param datum
+	 */
+	getDualAxesDomainPosition(datum) {
```

Comment:
doesn't this work regardless of whether you have a dual-axis setup or not? Maybe it should just be called `getDatumCorrespondingAxisPosition` or something like that

Reply:
maybe I'm misunderstanding what this method is doing, regardless the name is confusing